overlord/devicestate/handlers_serial.go

```diff
@@ -664,6 +664,12 @@ func (m *DeviceManager) doRequestSerial(t *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
+	// wait for ntp sync before trying to register the serial
+	maxWait := 10 * time.Minute
+	if !m.ntpSyncedOrWaitedLongerThan(maxWait) {
```

Comment:
Doesn't this put us in the same scenario as before? The retries would be postponed by the Ensure interval, so ~5 minutes anyway. Perhaps what we need is to subscribe to a signal from timedate1 and kick EnsureBefore when NTPSynchronized becomes true.

Reply:
Retry can be built with After to control that

Reply:
Right, we could make that more aggressive